database.go

```diff
@@ -149,12 +163,104 @@ func (db *Database) AddBinary(binaryContent []byte) *Binary {
 	return db.Content.Meta.Binaries.Add(binaryContent, WithKDBXv31Binary)
 }
 
+type BinarySet map[*Binary]bool
+type BinaryDuplicates map[*Binary]*Binary
+type BinaryUsages map[int][]BinaryReference
+
+func (db *Database) CleanupBinaries() int {
```

Comment:
This does not need to be exported. If we want this to be a convenience feature it should be called before encoding a file, but never manually

Reply:
Agree, but a little bit doubt about case when user doesn’t intend to change database, but can check if binaries can be optimized and clear it.

Reply:
If the user does not intent to change the database then encoding and writing the file can be skipped